src/charm.py

```diff
@@ -119,10 +118,12 @@ def _configure(self, _):
         except CharmConfigInvalidError:
             return
 
+        if self._charm_config.upf_mode == UpfMode.dpdk:
+            self._create_access_interface()
+            self._create_core_interface()
```

Comment:
Why not have this logic in the `self._network.configure()` like the rest of the network configuration?

Reply:
That would also require a lot of changes in the main callback of the charm. In the charm, before we run network configuration, we check whether the interfaces exist. With DPDK they don't; that would put us in a deadlock. Alternatively we could add another `if` to bypass the check if we're in DPDK mode, but it doesn't feel right. 

BTW, that's why I mentioned that the whole charm could use some refactoring ;)